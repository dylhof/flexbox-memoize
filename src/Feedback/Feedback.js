import React from 'react';

function Feedback(props) {
  return(  
    <div>
      <h3> {props.message} </h3>
      <p> property </p>
      <p> description </p>
      <button onclick={props.updateQuestion}> Next Question </button>
    </div>
    )
}

export default Feedback;