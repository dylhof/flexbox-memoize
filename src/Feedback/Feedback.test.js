import React from 'react';
import Feedback from './Feedback';
import { shallow } from 'enzyme';

const question = {
  "property": "flex-direction: row",
  "correctAnswer": "https://i.imgur.com/Xo1MD86.png",
  "answers": ["https://i.imgur.com/Xo1MD86.png", "https://i.imgur.com/JIJHnVf.png", "https://i.imgur.com/4y9cZDw.png", "https://i.imgur.com/Id0kLlR.png"],
  "info": "This establishes the main-axis, thus defining the direction flex items are placed in the flex container. Using row organizes the items from left to right.",
  "family-member": "Parent",
  "model": "Flexbox"
}

describe('Feedback', () => {
  it('should match the snapshot for a correct answer', () => {
    let wrapper = shallow(
      <Feedback 
        answerCorrect={true}
        currentQuestion={question}/>
    )
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot for an incorrect answer', () => {
    let wrapper = shallow(
      <Feedback 
        answerCorrect={false}
        currentQuestion={question}/>
    )
    expect(wrapper).toMatchSnapshot();
  })
})