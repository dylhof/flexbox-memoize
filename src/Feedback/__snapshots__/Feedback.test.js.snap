// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback should match the snapshot for a correct answer 1`] = `
<div
  className="feedback-div"
>
  <h3>
    Correct!
  </h3>
  <p
    className="feedback-property-p"
  >
     
    flex-direction: row
     
  </p>
  <p
    className="feedback-info-p"
  >
     
    This establishes the main-axis, thus defining the direction flex items are placed in the flex container. Using row organizes the items from left to right.
     
  </p>
  <img
    alt="flex-direction: row"
    className="answer-img"
    key="flex-direction: row"
    src="https://i.imgur.com/Xo1MD86.png"
  />
  <button
    className="next-btn"
  >
     Next Question 
  </button>
</div>
`;

exports[`Feedback should match the snapshot for an incorrect answer 1`] = `
<div
  className="feedback-div"
>
  <h3>
    Not Quite...
  </h3>
  <p
    className="feedback-property-p"
  >
     
    flex-direction: row
     
  </p>
  <p
    className="feedback-info-p"
  >
     
    This establishes the main-axis, thus defining the direction flex items are placed in the flex container. Using row organizes the items from left to right.
     
  </p>
  <img
    alt="flex-direction: row"
    className="answer-img"
    key="flex-direction: row"
    src="https://i.imgur.com/Xo1MD86.png"
  />
  <button
    className="next-btn"
  >
     Next Question 
  </button>
</div>
`;
